evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -754,7 +772,49 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> Handler for CheatcodeStackExecutor<'a
                 true,
                 true,
                 context,
-            )
+            );
+
+            if let Some(expected_revert) = expected_revert {
+                let final_res = match res {
+                    Capture::Exit((ExitReason::Revert(_e), data)) => {
+                        if data.len() >= 4 && data[0..4] == [8, 195, 121, 160] {
+                            // its a revert string
+                            let len = U256::from(&data[4 + 32..36 + 32]).as_usize();
+                            if data[4 + 32 + 32..4 + 32 + 32 + len] == *expected_revert {
```

Comment:
I'd do a `ethers::abi::decode([ParamType::String], data[4..])` instead of manually parsing the bytes, https://docs.rs/ethabi/latest/ethabi/fn.decode.html. 

If you don't want to do that, I'd suggest maybe abstracting the "decode length and everything else after the function selector" logic in a helper function that takes in bytes and returns a string or byte slice